### ISSUE TITLE
Cognitive Search & Open AI responds outside the data files uploaded. The answers are generated apart from the sources.

### ISSUE THREAD

**Issue by ArunEPRO** [Azure-Samples/azure-search-openai-demo#267]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Ask questions like what is Microsoft? It immediately responds by collecting information from microsoft.com. 

### Any log messages given by the failure
> No

### Expected/desired behavior
> I am expecting the answer to be, "I am sorry, I don't have any information about Microsoft and I can help you product information of xxx company". The data I uploaded do not have any information about Microsoft. I checked prompt prefix and it says clearly " If there isn't enough information below, say you don't know. Do not generate answers that don't use the sources below"

### OS and Version?
> Linux - App Service

### azd version?
azd version 1.0.1 (commit e0cd1aca716fa5d08704beade7dcc734fe68f5f1)

**Comment by itmilos** [comment]
Yes this is the case we need to implement system to prevent this from happening.

**Comment by ArunEPRO** [comment]
Have you implemented it in your system? If so how?

I checked my prompt and they are clear as I mentioned earlier.

**Comment by sofiangava** [comment]
Any update on how to ensure that the response is only from the data which has been supplied. The behaviour we have encountered is variable.

**Comment by kishorerv93** [comment]
Update the system message in approaches/crr to tell assistant to answer strictly from the sources below and do not generate responses from the GPT knowledge.

**Comment by itmilos** [comment]
> Update the system message in approaches/crr to tell assistant to answer strictly from the sources below and do not generate responses from the GPT knowledge.

I have tried this but it would still answer from GPT knowledge.

**Comment by kishorerv93** [comment]
Do you mind showing your system message?

**Comment by itmilos** [comment]
@kishorerv93 what do you mean by system message?

**Comment by ArunEPRO** [comment]
My prompt has this information - " If there isn't enough information below, say you don't know. Do not generate answers that don't use the sources below"

But it still responds outside the source.

**Comment by sagarsat** [comment]
I'm also facing this issue. 

Let me know if something needs to be changed.

System prompt I'm using:

You are an AI assistant designed to help technicians with their queries around product troubleshooting. Answer ONLY with the facts listed in the SOURCES below. IMPORTANT: DO NOT GENERATE ANSWERS THAT DO NOT USE THE SOURCES. AND ALSO DO NOT MAKE UP OR INFER ANY ADDITIONAL INFORMATION THAT IS NOT INCLUDED BELOW IN SOURCES. If asking a clarifying question to the user would help, ask the question. Each source has a name followed by colon and the actual information, always include the source name for each fact you use in the response. Use square brackets to reference the source, e.g. [info1.txt]. Don't combine sources, list each source separately, e.g. [info1.txt][info2.pdf]. SOURCES: {_sources}

**Comment by pamelafox** [comment]
Thank you for reporting this grounding issue back in June 2023. I'm closing this as obsolete because the project has made significant improvements to address this concern:

**What's Changed:**

1. **Evaluation Framework**: We've added a comprehensive evaluation system (December 2024) with "groundedness" metrics that specifically measure how well answers stick to provided sources. See our [evaluation guide](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/docs/evaluation.md).

2. **Model Upgrades**: The default model has evolved from gpt-35-turbo → gpt-4o-mini → gpt-4.1-mini, with each generation showing improved instruction-following and grounding behavior. **Most significantly, GPT-5 (released in 2025) demonstrates substantially better grounding capabilities** and is much more reliable at sticking to provided sources in RAG applications.

3. **Better Tooling**: We now provide extensive documentation on [improving answer quality](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/docs/customization.md#improving-answer-quality) with specific guidance on tuning prompts and evaluating grounding.

**Important Context:**

While these improvements significantly reduce instances of the model generating answers outside the provided sources, it's important to understand that **perfect grounding cannot be 100% guaranteed** with LLMs. This is a known limitation of the underlying technology. However, **GPT-5 has made major strides in this area**, providing much more consistent grounding behavior compared to earlier models. For more details on GPT-5's RAG capabilities, see: https://blog.pamelafox.org/2025/08/gpt-5-will-it-rag.html

The evaluation tools we've added allow you to measure groundedness quantitatively and tune your system to minimize these occurrences.

If you're experiencing grounding issues with the current version of the application, I encourage you to:
- **Try GPT-5** - It offers significantly better grounding and source adherence
- Use the evaluation framework to measure groundedness metrics
- Review the [customization guide](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/docs/customization.md) for prompt tuning strategies

Please feel free to open a new issue if you encounter specific grounding problems with the current version of the application.
